src/main/scala/org/clulab/timenorm/scfg/Evaluator.scala

```diff
@@ -0,0 +1,193 @@
+      // If this is a timex (is not a doc separator):
+      if (timex != "") {
+        println(timex)
+        // If this is the first timex in a doc, consider it a DCT
+        if (dct_timex == "") {
+          dct_timex = timex
+        }
+        // Normalize the timex and append the normalization
+        var value = normalize(parser, timex, dct_timex)
+        norm_list += value
+      }
+      // If this is a doc separator, empty the DCT timex and append ""
+      else {
+        dct_timex = ""
+        norm_list += ""
+      }
+    }
+    return norm_list.toList
+  }
+
+
+  def normalize(parser:TemporalExpressionParser, timex:String, dct_timex:String) : String = {
+    /**Normalizes a timex according to the parser and the DCT timex.
+    DCTs are normalized with respect to themselves*/
+
+    // Process the DCT depending on the presence of a time reference
+    val pattern = "T".r
+    val anchor = pattern.findFirstIn(dct_timex) match {
+      // Get the anchor timespan if time is specified
+      case Some(_) =>
+        val dct = dct_timex.split("T")
+        val date = dct(0).split("-")
+        val time = dct(1).split(":")
+        val year = date(0).toInt
+        val month = date(1).toInt
+        val day = date(2).toInt
+        val hours = time(0).toInt
+        val minutes = time(1).toInt
+        val seconds = time.length match {
+          case 3 => time(2).toInt
+          case _ => 0
+        }
+        TimeSpan.of(year, month, day, hours, minutes, seconds)
+
+      // Get the anchor timespan if time is not specified
+      case None =>
+        val dct = dct_timex.split("-")
+        val year = dct(0).toInt
+        val month = dct(1).toInt
+        val day = dct(2).toInt
+        TimeSpan.of(year, month, day)
+    }
+
+    // Parse the timex with respect to its anchor
+    parser.parse(timex, anchor) match {
+      // If the parser fails, return an empty string as normalization
+      case Failure(temporal) =>
+        return "-"
+      // If the parser successes, return the normalization of the timex
+      case Success(temporal) =>
+        return temporal.timeMLValue
+    }
+  }
+
+
+  def compare_and_write(out_file:String, timex_list:List[String],
+    gold_list:List[String], norm_list:List[String]) : (Int, Int) = {
+    /**Writes the results on the output file, in "{timex}\t{gold}\t{norm}"
+    format, and counts the number of timexes and correct normalizations*/
+
+    // Create the output writer
+    val writer = new PrintWriter(new File(out_file))
+
+    var gold_counter = 0
+    var norm_counter = 0
+
+    // Iterate over timex list and get each timex, gold and norm set
+    for (i <- 0 to timex_list.length - 1) {
+      val timex = timex_list(i)
+      val gold = gold_list(i)
+      val norm = norm_list(i)
+      println(s"${timex}\t${gold}\t${norm}")
+
+      // If this is a timex, write the data and sum a gold value
+      if (timex != "") {
+        writer.write(s"${timex}\t${gold}\t${norm}\n")

```

Comment:
Since it is a PrintWriter, println() would work well.

Reply:
OK, thanks!